### DISCUSSION TITLE
How to get Ogen to generate a map[string]any type?

### DISCUSSION THREAD

**Discussion by veqryn** [ogen-go/ogen#1311] [Q&A]
Right now in my OpenAPI yaml, we have a component with a field that is marked as `type: object`.
In all other OpenAPI code generators, this gets turned into `map[string]any`, letting us use this field for any key-value pairs (this is our desired goal).

How should we define our OpenAPI yaml so that we either get Ogen to generate a map[string]any, or get it to generate something effectively equivalent (allows arbitrary key-value pairs of anything).

**Comment by ghost** [comment]
You need to explictly define `additionalProperties: true`.

https://ogen.dev/docs/types/map#free-form-map

```yaml
type: object
additionalProperties: true
```

**Comment by veqryn** [comment]
@tdakkota 
Hm... that resulted in generating a `map[string]jx.Raw`, where jx.Raw is a `[]byte` under the hood.
That is pretty hard to use.  I'd have to manually json marshal every single property, rather than just setting them all and letting the whole thing be json marshalled.

Is there any option for a `map[string]interface{}` / `map[string]any` ?
Or even just an option for a `any`, where I can then use my own map?

**Reply by ghost** [reply]
No, currently there is no way to make ogen generate `map[string]any`. Feel free to fill in a feature request issue.

**Reply by veqryn** [reply]
thx, done: https://github.com/ogen-go/ogen/issues/1313
